### PR TITLE
[eight] Backport of my prior updates in Git (Pull #1649)

### PR DESCRIPTION
**Pull request by jaredgrubb**
This backports from the change merged in Pull #1649:
 * move a couple functions to inlineCallbacks
 * remove unused variable
 * also fix a pylint complaint

### CODE REVIEW

File: master/buildbot/test/unit/test_plugins.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ class ITestInterface(IPlugin):
     """
     test interface
     """
-    def hello(name):
```
Comment (jaredgrubb) [inline master/buildbot/test/unit/test_plugins.py]: This was suggested by pylint, saying the member function had no 'self'.

Comment (djmitche) [inline master/buildbot/test/unit/test_plugins.py]: Generally interfaces don't list `self`, but that confuses pylint.  It doesn't hurt to have `self`, so this is fine.

File: master/buildbot/steps/source/git.py [hunk 2]
```diff
@@ -570,35 +567,32 @@ def applyAlready(res):
         d.addCallback(applyAlready)
         return d
 
+    @defer.inlineCallbacks
     def _sourcedirIsUpdatable(self):
         if self.slaveVersionIsOlderThan('listdir', '2.16'):
-            d = self.pathExists(self.build.path_module.join(self.workdir, '.git'))
+            git_path = self.build.path_module.join(self.workdir, '.git')
```
Comment (sa2ajj) [inline master/buildbot/steps/source/git.py]: do we use `git_path` anywhere else?

File: master/buildbot/steps/source/git.py [hunk 3]
```diff
@@ -570,35 +567,32 @@ def applyAlready(res):
...
-                elif len(files) > 0:
-                    return "clobber"
-                else:
-                    return "clone"
-            d.addCallback(checkWithListdir)
-        return d
+        cmd = buildstep.RemoteCommand('listdir',
+                                      {'dir': self.workdir,
+                                       'logEnviron': self.logEnviron,
+                                       'timeout': self.timeout, })
```
Comment (sa2ajj) [inline master/buildbot/steps/source/git.py]: s/, //  ?

File: master/buildbot/test/unit/test_plugins.py (hunk 1 above)
Comment (sa2ajj) [inline master/buildbot/test/unit/test_plugins.py]: Do we need this function at all?

Comment (sa2ajj) [inline master/buildbot/test/unit/test_plugins.py]: Yes, we do and it was _I_ who wrote it :(

File: master/buildbot/steps/source/git.py (hunk 2 above)
Comment (jaredgrubb) [inline master/buildbot/steps/source/git.py]: No, it's not. the line was getting a bit long and I wanted to split it... Plus, giving things a name can help readability and make the lines easier to read.

Comment (sa2ajj) [inline master/buildbot/steps/source/git.py]: Sometimes it helps, and sometimes it does not.

/me is not going to argue here :)

**Comment by jaredgrubb** [comment]
Adjusted the comma as requested.

**Pull request merged**